assets/javascripts/discourse/initializers/discourse-events.js

```diff
@@ -271,42 +271,35 @@ export default {
           setupController(controller, model) {
             const start = this.get("start");
             const end = this.get("end");
+            let initialDateRange;
 
             if (start || end) {
-              let initialDateRange = {};
+              initialDateRange = {};
               if (start) {
                 initialDateRange["start"] = start;
               }
               if (end) {
                 initialDateRange["end"] = end;
               }
-              this.controllerFor("discovery/topics").setProperties({
-                initialDateRange,
-              });
             }
 
-            this._super(controller, model);
-          },
-
-          renderTemplate(controller, model) {
             // respect discourse-layouts settings
             const global = siteSettings.layouts_list_navigation_disabled_global;
             const catGlobal =
               model.category &&
               model.category.get("layouts_list_navigation_disabled_global");
-            if (!global && !catGlobal) {
-              if (this.routeName.indexOf("Category") > -1) {
-                this.render("navigation/category", {
-                  outlet: "navigation-bar",
-                });
-              } else {
-                this.render("navigation/default", { outlet: "navigation-bar" });
-              }
-            }
-            this.render("discovery/calendar", {
-              outlet: "list-container",
-              controller: "discovery/topics",
+            const showNavigation = !global && !catGlobal;
+
+            controller.setProperties({
+              initialDateRange,
+              showNavigation,
             });
+
+            this._super(controller, model);
+          },
+
+          renderTemplate() {
+            this.render("discovery/calendar");
```

Comment:
This isn't strictly related to the discovery refactoring, but I think it's worth noting that `this.render()` is deprecated in Ember and is removed in Ember 4.x+.  (indeed, this is the reason core is refactoring the discovery routes to remove named outlets and `this.render()` calls)

Right now the deprecation is silenced in Discourse core, so it won't appear in the console.

I think in this case, you're not actually using a 'named outlet', so you should be able to resolve the issue fairly easily by removing this `renderTemplate()` function and instead setting

```
templateName: "discovery/calendar"
```
on the controller. That is still supported into Ember 4.x and beyond.

(see https://deprecations.emberjs.com/v3.x#toc_route-render-template for more info)

Reply:
> Right now the deprecation is silenced in Discourse core, so it won't appear in the console.

(I just updated the `discovery-named-outlets` branch of Discourse so that the relevant deprecation is no longer silenced)

Reply:
Thanks david 👍

Reply:
@davidtaylorhq Experimenting with this this morning, I couldn't get `templateName` to work using the plugin api (i.e. `modifyClass`). Creating an entirely separate route using `buildCategoryRoute` does work however

https://github.com/paviliondev/discourse-events/commit/3da7ef5c08e21faf664a6cba92570fe80923bbd2

Reply:
> I couldn't get templateName to work using the plugin api (i.e. modifyClass).

Ah yes, that'll be because 'native class fields' will always take precedence over properties set via Ember's legacy object-literal authoring format (which modifyClass currently uses). Moving it to a dedicated class definition seems like a good solution, and a good general improvement anyway 👌